src/cached_object_store.rs

```diff
@@ -31,6 +35,13 @@ pub(crate) struct CachedObjectStore {
     object_store: Arc<dyn ObjectStore>,
     pub(crate) part_size_bytes: usize, // expected to be aligned with mb or kb
     pub(crate) cache_storage: Arc<dyn LocalCacheStorage>,
+    // ghost queue is used to track identifiers of elements that are only visited once
+    // we use this to try to avoid elements that are accessed only once from actually
+    // being written to the disk cache. This helps to keep popular elements in the disk
+    // cache as much as possible, and also improves performance and extends LocalCacheStorage(SSD...)
+    // lifetime at the same time.
+    ghost: Arc<Mutex<LruCache<Path, ObjectMeta>>>,
+    max_cache_size_bytes: Option<usize>, // Used for ghost queue to self-tuning
```

Comment:
this config is not only used for ghost queue, right?

it might be better to put all its usages in the comment.